tasks/configure.yml

```diff
@@ -0,0 +1,37 @@
+---
+- name: Create dhcp config for host
+  template: src=node_dhcp.conf.j2 dest="/etc/dhcp/dhcpd.d/{{ inventory_hostname_short }}.conf"
+  when: use_dhcp or use_pxe
+
+- name: Create pxe boot data (read by boot.py)
+  template: src=node_pxe.conf.j2 dest="/var/www/provision/nodes/{{ inventory_hostname_short }}.conf"
+  when: use_pxe
+
+- name: Create kickstart file
+  template: src=kickstart.j2 dest="/var/www/html/kickstart/{{ inventory_hostname_short }}.ks"
+  when: use_pxe and use_kickstart
+
+- name: Find all the configs in dhcpd.d directory
+  run_once: true
+  find:
+    paths: "/etc/dhcp/dhcpd.d/"
+    patterns: "*.conf"
+  register: configs
+ 
+- name: Clear hosts config
```

Comment:
Not a big deal, but this and the next task could be quite easily replaced by a template like this. The main benefits would be a more standard way of doing it, and avoiding unnecessary "changed" tasks, and service restarts.

 - name: Create dhcp include file
    run_once: true
    template: src=dhcp_include.j2 dest="/etc/dhcp/dhcpd_hosts.conf"
    notify: restart dhcpd